### DISCUSSION TITLE
Development history

### DISCUSSION THREAD

**Discussion by GenEugene** [GenEugene/GETools#12] [Show and tell]
Here is a comparison of how Overlappy script changed over time. Today it is a part of GETools plugin.

![Screenshot 2023-11-12 175154](https://github.com/GenEugene/GETools/assets/31050379/09029eee-76c5-4481-9f97-f1dd0d8551bd)

**Comment by GenEugene** [comment]
Here is a new update of version 0.0.11
![collage](https://github.com/GenEugene/GETools/assets/31050379/82e52945-d66b-4964-a5d2-9d9d3ad8447b)

**Reply by apriore** [reply]
Hi GenEugene, congratulations on your tools!
I've been using overlappy for a long time and I think it's amazing. I tried the new version of overlappy and was curious to try the center of the mass script, but I don't understand how they work!
Would it be possible to have a quick tutorial on the wiki?

With overlappy I can't get the overlap animations to work on rotation, even if I increase the values a lot there is no secondary animation. I'm using maya 2023.

Thank you

**Reply by GenEugene** [reply]
Hi Apriore. Thanks for using my scripts, I hope it useful for you.
I still plan tutorials for all features, because some of them could be very confusing first time. I use them all in my work and it's save my time a lot.

Of course I can add some using scenarios to WIKI for these tools, but here short examples:

Overlappy. In the earliest version you just selected objects in row and press bake button, distances between objects was compiled automatically and rotation worked from start. But usually it was so inconsistent and you always needed to create a tip locator for last object to bake rotation to it.
In new version I reworked all approach and you need to specify offset axis and distance for use rotation overlaps. In the bottom part of script you can find Offset settings. So, this is the key element. 
Yes, it's confusing, I need to figure out, how to do it more intuitive and of course I need to make a video tutorial.

**Reply by GenEugene** [reply]
About Center of Mass. It's pretty simple tool and it helps me a lot to understand my masses more. Also I can make some rebaking for counter animation to compensate some center of mass movements or directly animate center of mass by other objects.

Also I need to make video tutorial with examples of use scenario, but main idea is just Create COM object by script button and assign some objects to this COM with weights.

For example:
If I need to setup COM on my character, I will use predefined weights from WEIGHTS section. Simply select head joint/control and press button with head weight (or any button with number, you can change it later). Then select chest joint/control and press CHEST button. And the same process again and again.

In result you will have a Center Of Mass object that will stay approximately on the middle of all weighted objects, depends on how much weight you used before.

This weight you can animate so imitate weight lifting or putting. It's a lot of potential in this tool

**Comment by apriore** [comment]
Hi GenEugene,
Thanks for the quick response and support!

Your scripts are really very helpful to me!
Thanks for all the work you are doing!
Andrea

**Comment by apriore** [comment]
I followed your advice for overlap and it works perfectly!

Now I'll do some tests with the center of mass.
A thousand thanks
Andrea

**Reply by GenEugene** [reply]
Added short how to use instruction to Overlappy WIKI https://github.com/GenEugene/GETools/wiki/Overlappy
If it's unclear or you have ideas how to improve description, please let me know. I try to use all feedback

**Comment by apriore** [comment]
Hi GenEugene, thank you so much for the great support!
I tried Center of mass with your instructions and it works great, it's a tool with great potential.

Thanks also for the extra info about overlappy on wiki, everything is clear now!
Thank you so much!
Andrea

**Reply by GenEugene** [reply]
You are welcome 😁. Ask me if something unclear

**Comment by GenEugene** [comment]
Version 1.0.0
Added an important window with different buttons. They allow you to create shelf buttons with specific functions from script, so you don't need to open GETools each time, just extract what you use.

Also you can create a hotkey, just need to copy code from button and put to hotkey window in Python mode

![image](https://github.com/GenEugene/GETools/assets/31050379/ad38d2fc-7f42-418a-9387-abd17c7a6af0)

**Comment by GenEugene** [comment]
![Collage_v1 0 0](https://github.com/GenEugene/GETools/assets/31050379/37659763-3760-469d-9afb-7572be9eece7)

**Comment by apriore** [comment]
Hi GenEugene,
sorry for the delay, I was on holiday!

Happy New Year! ☺
Thanks for the updates and the great support of the Tools.

Andrea

**Comment by GenEugene** [comment]
![Collage_v1 5 1](https://github.com/user-attachments/assets/5417a9d7-e5f3-4233-8a4b-ad5f96e6312d)
